lib/kube/proxy/portforward_test.go

```diff
@@ -213,3 +219,171 @@ type portForwarder interface {
+	// unblock the targetConn.waitForClose() call otherwise it will block
+	// forever.
+	targetConn.waitForClose()
+	// Make sure all streams are removed from the connection.
+	require.Empty(t, sourceConn.streams)
+	// Make sure all streams are removed from the connection.
+	require.Empty(t, targetConn.streams)
+	require.True(t, sourceConn.wereStreamsClosed())
+	require.True(t, targetConn.wereStreamsClosed())
+}
+
+var _ httpstream.Stream = (*fakeSPDYStream)(nil)
+
+type fakeSPDYStream struct {
+	closed     bool
+	headers    http.Header
+	identifier uint32
+}
+
+func (f *fakeSPDYStream) Read(p []byte) (n int, err error) {
+	return 0, io.EOF
+}
+
+func (f *fakeSPDYStream) Write(p []byte) (n int, err error) {
+	return len(p), nil
+}
+
+func (f *fakeSPDYStream) Headers() http.Header {
+	return f.headers
+}
+
+func (f *fakeSPDYStream) Reset() error {
+	return nil
+}
+
+func (f *fakeSPDYStream) Identifier() uint32 {
+	return f.identifier
+}
+
+func (f *fakeSPDYStream) Close() error {
+	f.closed = true
+	return nil
+}
+
+var _ httpstream.Connection = (*fakeSPDYConnection)(nil)
+
+type fakeSPDYConnection struct {
+	count        int
+	streams      map[uint32]*fakeSPDYStream
+	streamsSlice []*fakeSPDYStream
+	closed       chan bool
+	once         sync.Once
+}
+
+func newfakeSPDYConnection() *fakeSPDYConnection {
+	return &fakeSPDYConnection{
+		streams: make(map[uint32]*fakeSPDYStream),
+		closed:  make(chan bool),
+	}
+}
+
+// CreateStream creates a new Stream with the supplied headers.
+func (f *fakeSPDYConnection) CreateStream(headers http.Header) (httpstream.Stream, error) {
+	newHeader := http.Header{}
+	for k, v := range headers {
+		newHeader.Set(k, v[0])
+	}
+	f.count++
+	identifier := uint32(f.count)
+	stream := &fakeSPDYStream{identifier: identifier, headers: newHeader}
+	f.streamsSlice = append(f.streamsSlice, stream)
+	f.streams[identifier] = stream
+	return stream, nil
+}
+
+// Close resets all streams and closes the connection.
+func (f *fakeSPDYConnection) Close() error {
+	f.once.Do(func() {
+		close(f.closed)
+	})
+	return nil
+}
+
+// CloseChan returns a channel that is closed when the underlying connection is closed.
+func (f *fakeSPDYConnection) CloseChan() <-chan bool {
+	return f.closed
+}
+
+// SetIdleTimeout sets the amount of time the connection may remain idle before
+// it is automatically closed.
+func (f *fakeSPDYConnection) SetIdleTimeout(_ time.Duration) {
+}
+
+// RemoveStreams can be used to remove a set of streams from the Connection.
+func (f *fakeSPDYConnection) RemoveStreams(streams ...httpstream.Stream) {
+	for _, stream := range streams {
+		if stream == nil {
+			continue
+		}
+		delete(f.streams, stream.Identifier())
+	}
+	// if there are no streams left, close the connection so the test can exit
+	if len(f.streams) == 0 {
+		f.Close()
+	}
+}
+
+func (f *fakeSPDYConnection) waitForClose() {
+	<-f.closed
+}
+
+func (f *fakeSPDYConnection) wereStreamsClosed() bool {
```

Comment:
Suggestion: I would go with `streamsClosed` or maybe `isStreamsClosed`, out of habit most likely.